lib/temporal/activity/context.rb

```diff
@@ -84,6 +107,28 @@ def heartbeat(details = nil)
         self
       end
 
+      # Like heartbeat, but a subclass of ActivityInterrupted error will be raised if the
+      # activity has been canceled, start-to-close timeout exceeded, or if the worker is
+      # shutting down. This flag defaults to false, in which these states can be detected by
+      # inspecting the heartbeat response for cancelation, or calling timed_out? or shutting_down?
+      # methods.
+      def heartbeat_interrupted(details = nil)
```

Comment:
nit: `heartbeat!`? Similar to Rails' `save!` and `update!` which raise in case of a failure